File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int l = 0;

        for (int r = 1; r < prices.length; r++) {
            if (prices[r] < prices[l]) l = r;
            else maxProfit = Math.max(prices[r] - prices[l], maxProfit);
        }

        return maxProfit;
    }
}

File: Data Structures & Algorithms/network-delay-time/submission-0.py
class Solution:
    def networkDelayTime(self, times: List[List[int]], n: int, k: int) -> int:
        adj = {}

        for i in range(1, n + 1):
            adj[i] = []

        for s, d, w in times:
            adj[s].append([d, w])

        shortest = {}
        minHeap = [[0, k]]

        while minHeap:
            w1, n1 = heapq.heappop(minHeap)
            if n1 in shortest:
                continue

            shortest[n1] = w1

            for n2, w2 in adj[n1]:
                if not n2 in shortest:
                    heapq.heappush(minHeap, [w2 + w1, n2])
        
        if len(shortest) == n:
            return max(shortest.values())
        
        return -1